.circleci/config/jobs/website-docker-image.yml

```
@@ -10,14 +10,17 @@ steps:
         # There is an edge case that would cause an issue here - if dependencies are updated to an exact copy
         # of a previous version, for example if packge-lock.json is reverted, we need to manually push the new
         # image to the "latest" tag
-        IMAGE_TAG=$(cat website/Dockerfile website/package-lock.json | sha256sum | awk '{print $1;}')
-        echo "Using $IMAGE_TAG"
-        if ! curl https://hub.docker.com/v2/repositories/hashicorp/nomad-website/tags/$IMAGE_TAG -fsL > /dev/null; then
-          cd website/
-          docker build -t hashicorp/nomad-website:$IMAGE_TAG .
-          docker tag hashicorp/nomad-website:$IMAGE_TAG hashicorp/nomad-website:latest
-          docker login -u $WEBSITE_DOCKER_USER -p $WEBSITE_DOCKER_PASS
-          docker push hashicorp/nomad-website
-        else
-          echo "Dependencies have not changed, not building a new website docker image."
+        # Ignore job if running an enterprise build
+        if ! [ "$NOMAD_ENT_BUILD" = "true" ]; then
```

I think this looks ok but where does this env var get set? I don't see it on any of the .circleci configs.

---

It's set in CCI enterprise

---

You can filter out all non-nomad-OSS repositories with something like this:

```suggestion
        if [ "$CIRCLE_REPOSITORY_URL" == "git@github.com:hashicorp/nomad.git" ]; then
```

Also, to avoid nesting, I would consider rearranging conditionals so we can bring the skipping conditions together and avoid nesting like the following:

```bash
IMAGE_TAG=$(cat website/Dockerfile website/package-lock.json | sha256sum | awk '{print $1;}')
echo "Using $IMAGE_TAG"
if [ "$CIRCLE_REPOSITORY_URL" != "git@github.com:hashicorp/nomad.git" ]; then
  echo Not Nomad OSS repo
elif curl https://hub.docker.com/v2/repositories/hashicorp/nomad-website/tags/$IMAGE_TAG -fsL > /dev/null; then
  echo "Dependencies have not changed, not building a new website docker image."
else
  cd website/
  ....
fi
```

---

That's a much better approach, thanks!